File: C++/Valid Parentheses.cpp
#include<iostream>
#include<stack>

bool isValid(string s)
    {
        stack<char> st;
        for(int i=0;i<s.size();i++)
            {
                if(s[i]=='(')
                    {
                        st.push(s[i]);
                    }
                else if(s[i]=='[')
                    {
                        st.push(s[i]);
                    }
                else if(s[i]=='{')
                    {
                        st.push(s[i]);
                    }
                else if(s[i]==')')
                    {
                        if(st.empty()) return false;
                        if(st.top()!='(') return false;
                        st.pop();
                    }
                else if(s[i]==']')
                    {
                        if(st.empty()) return false;
                        if(st.top()!='[') return false;
                        st.pop();
                    }
                else if(s[i]=='}')
                    {
                        if(st.empty()) return false;
                        if(st.top()!='{') return false;
                        st.pop();
                    }
            }
        if(st.empty()) return true;
        else return false;
    }

File: Python/Valid-Parentheses.py
#!/usr/bin/env python

def isValid(self, s):
    length = len(s)
    seq = []
    for i in range(length):
        if s[i] == '{' or s[i] == '[' or s[i] == '(':
            seq.append(s[i])
        else:
            if s[i] == '}':
                if len(seq) >=1 and seq[len(seq) - 1] == '{':
                    seq.pop()
                else:
                    return False
            if s[i] == ']':
                if len(seq) >=1 and seq[len(seq) - 1] == '[':
                    seq.pop()
                else:
                    return False
            if s[i] == ')':
                if len(seq) >=1 and seq[len(seq) - 1] == '(':
                    seq.pop()
                else:
                    return False
    if len(seq) != 0:
        return False
    return True